crypto/salsa20/xmm6int/salsa20_xmm6int-sse2.h

```diff
@@ -3,7 +3,7 @@
 
 #include "../stream_salsa20.h"
 #include <sodium/crypto_stream_salsa20.h>
-#ifndef ANDROID
+#if !defined(ANDROID) && !defined(iOS)
```

Comment:
This should read
```c
#if __SSE2__
```
instead, using inbuilt compiler support to detect whether the SSE2 instruction is present, rather then relying on explicit build flags/defines.